zipline/data/data_portal.py

```diff
@@ -916,6 +916,8 @@ def _get_minute_window_for_equities(
             return np.full((len(minutes_for_window), len(assets), np.nan))
 
         num_minutes = len(minutes_for_window)
+        start_date = normalize_date(minutes_for_window[0])
```

Comment:
i.e. we would benefit from a function where the interface is `intraday_value_up_to_dt(assets, field, dt)`, which can use a cached value of `intraday_value_up_to_dt(assets, field, dt - 1)` to answer the question of 'what is the lowest low to this point' etc.

e.g.

```
def intraday_value_up_to_dt(self, asset, field, dt):
    try:
        prev = self.intraday_cache[(asset, field, dt - 1)]
        value_at_dt = # read from prefetched window
        if field == 'low': 
            value = min(value_at_dt, prev)
    except KeyError:
        # get values to dt, except for open and close, which just need the first and last
        if field == 'low':
           value = values_to_dt.min()
    self._intraday_cache[(asset, field, dt)] = value
    return value   
```